### ISSUE TITLE
i18n doesn't define i18n locale on server (production)

### ISSUE THREAD

**Issue by lonagi** [nuxt-modules/i18n#2203]
### Environment

Nuxi 3.6.1                                                                                                   2:15:21 PM

------------------------------
- Operating System: Darwin
- Node Version:     v19.2.0
- Nuxt Version:     3.6.1
- Nitro Version:    2.5.2
- Package Manager:  npm@8.19.3
- Builder:          vite
- User Config:      devtools, typescript, telemetry, modules, ssr, experimental, runtimeConfig, css, components, plugins, i18n
- Runtime Modules:  @nuxtjs/i18n@8.0.0-beta.12, @nuxtjs/tailwindcss@6.8.0
- Build Modules:    -
------------------------------

### Reproduction

https://stackblitz.com/edit/github-7xfigw?file=nuxt.config.ts

### Describe the bug

When I run `npm run dev` all works. I verify it according to view-source in Google Chrome. I see that nuxt gives me needed strings from i18n.

But when I run
```
npm ci && \
npm cache clean --force && \
npm run build
```

and

`node .output/server/index.mjs`

i18n works only on client, but "view-source" doesn't see any translations. I think definei18nLocale works, because If I put console.log there, then it prints text but only when I run first build commands. It sees language (I put "en" always to test), and messages from server, but doesn't apply it to Nuxt App idk.

And in node runtime when I go to page I see error on node console (see logs).

### Additional context

I fetch messages from API backend

### Logs

```shell
Listening http://[::]:3000
[intlify] The message format compilation is not supported in this build. Because message compiler isn't included. You need to pre-compilation all message format. So translate function return 'browse'.
```

**Comment by lonagi** [comment]
It seems that this problem was here: **i18n issues when running "node .output/server/index.mjs", while "nuxt dev" and "nuxt preview" works fine #1833**

You can close. But I still cannot understand what to do :)

**Comment by zzzhim** [comment]
我之前遇到了和你相同的问题，这是我的配置文件，希望能够帮助你
I had the same problem as you before, here is my config file, hope it can help you

```ts
// nuxt.config.ts
{
modules: [
    "@nuxtjs/i18n",
],

i18n: {
    vueI18n: "./i18n.config.ts",
    detectBrowserLanguage: {
      useCookie: true,
      cookieKey: "i18n_redirected",
      redirectOn: "root", // recommended
    },
    locales: [
      {
        code: "en",
        // file: "en/index.ts",
        iso: "en-US",
      },
      // {
      //   code: 'zh',
      //   iso: 'zh-CN'
      // },
      // {
      //   code: 'zh_hk',
      //   file: '/zh_hk/index.ts',
      //   iso: 'zh-HK'
      // },
      {
        code: "indonesia",
        // file: "indonesia/index.ts",
        iso: "id",
      },
      // {
      //   code: 'VI',
      //   file: 'VI/index.ts',
      //   iso: 'vi-VN'
      // },
      // {
      //   code: 'in_en',
      //   file: 'in_en/index.ts',
      //   iso: 'en-IN'
      // },
      // {
      //   code: 'ph',
      //   file: 'ph/index.ts',
      //   iso: 'en-PH'
      // },
      // {
      //   code: 'mys',
      //   file: 'mys/index.ts',
      //   iso: 'en-MY'
      // },
      {
        code: "pt",
        // file: "pt/index.ts",
        iso: "pt-BR",
      },
    ],
    defaultLocale: "en",
  }
{
```

```ts
// i18n.config.ts
import en from "./locales/en/index";
import indonesia from "./locales/indonesia/index";
import pt from "./locales/pt/index";

export default {
  lazy: false,
  locale: "en",
  fallbackLocale: "en",
  messages: {
    en,
    indonesia,
    pt,
  },
};

```

```json
{
  "private": true,
  "version": "0.0.3",
  "scripts": {
    "dev": "cross-env WEB_NODE_ENV=dev nuxt dev",
    "test": "cross-env WEB_NODE_ENV=test nuxt dev",
    "prev": "cross-env WEB_NODE_ENV=prev nuxt dev",
    "prod": "cross-env WEB_NODE_ENV=prod nuxt dev",
    "test_build": "cross-env WEB_NODE_ENV=test nuxt build",
    "prep_build": "cross-env WEB_NODE_ENV=prev nuxt build",
    "prod_build": "cross-env WEB_NODE_ENV=prod nuxt build",
    "generate": "nuxt generate",
    "preview": "nuxt preview",
    "postinstall": "nuxt prepare"
  },
  "type": "module",
  "devDependencies": {
    "@nuxt/image-edge": "^1.0.0-27968280.9739e4d",
    "@nuxtjs/i18n": "^8.0.0-beta.12",
    "@types/aos": "^3.0.4",
    "@types/node": "^18.15.5",
    "@vite-pwa/nuxt": "^0.0.7",
    "cross-env": "^7.0.3",
    "nuxt": "^3.6.2",
    "nuxt-seo-kit": "^1.3.8",
    "sass": "^1.59.3"
  },
  "dependencies": {
    "@nuxtjs/device": "^3.1.0",
    "@pinia/nuxt": "^0.4.7",
    "@vueuse/core": "^9.13.0",
    "@vueuse/nuxt": "^9.13.0",
    "aos": "^2.3.4",
    "dayjs": "^1.11.7",
    "pinia": "^2.0.33"
  }
}
```

修改pm2配置文件  `ecosystem.config.js` 为 `ecosystem.config.cjs`
Modify the pm2 configuration file `ecosystem.config.js` to `ecosystem.config.cjs`

执行 `pm2 start ./ecosystem.config.cjs --env=prep`
Execute `pm2 start ./ecosystem.config.cjs --env=prep`

**Comment by kazupon** [comment]
Thank you for your reporting!

This issue will be fixed with #2199 at near the future.
Please a little wait. 🙏
